dask_kubernetes/operator/controller/controller.py

```diff
@@ -549,15 +560,13 @@ async def daskworkergroup_replica_update(
     # Replica updates can come in quick succession and the changes must be applied atomically to ensure
     # the number of workers ends in the correct state
     async with worker_group_scale_locks[f"{namespace}/{name}"]:
-        current_workers = len(
-            await kr8s.asyncio.get(
-                "deployments",
-                namespace=namespace,
-                label_selector={"dask.org/workergroup-name": name},
-            )
+        current_workers = await kr8s.asyncio.get(
+            "deployments",
+            namespace=namespace,
+            label_selector={"dask.org/workergroup-name": name},
```

Comment:
Instead of listing all the worker deployments maybe we just want to list the Pods and use the field selector to get the ones that are Pending?

```python
current_workers = await kr8s.asyncio.get(
    "pods", 
    namespace=namespace, 
    label_selector={"dask.org/workergroup-name": name},
    field_selector="status.phase=Pending",
)
```

Then use the `dask.org/object-name` label (which I would maybe rename to `dask.org/component-name`) to delete the Deployment object later.

Reply:
> Instead of listing all the worker Deployments maybe we just want to list the Pods and use the field selector to get the ones that are Pending?

Couldn't that cause a time of check time of use bug?

Changing `dask.org/object-name` -> `dask.org/component-name` seems good to me

Reply:
> Couldn't that cause a time of check time of use bug?

Only if you delete the Pods sequentially. You could delete them concurrently in a task group.

```python
async with anyio.create_task_group() as tg:
    for pod in pods:
        tg.call_soon(pod.delete)
```